examples/capture/main.rs

```diff
@@ -164,12 +167,39 @@ fn main() {
     #[cfg(not(target_arch = "wasm32"))]
     {
         env_logger::init();
-        futures::executor::block_on(run());
+        futures::executor::block_on(run("red.png"));
     }
     #[cfg(target_arch = "wasm32")]
     {
         std::panic::set_hook(Box::new(console_error_panic_hook::hook));
         console_log::init().expect("could not initialize logger");
-        wasm_bindgen_futures::spawn_local(run());
+        wasm_bindgen_futures::spawn_local(run("red.png"));
     }
 }
+
+
+#[cfg(test)]
+mod tests{
+    use super::*;
+    use std::io::Read;
+    use std::fs::remove_file;
+
+    #[test]
+    fn ensure_generated_png_matches_expected(){
+        futures::executor::block_on(assert_generated_png_matches_expected(
+            "examples/capture/test_output.png",
+            "examples/capture/screenshot.png"));
+    }
+
+    async fn assert_generated_png_matches_expected(test_file_name: &str, expected_file_name: &str) {
```

Comment:
Let's instead compare the data we read back from `wgpu` (basically, raw pixels). There is no need for the data to go PNG encoding and saving to a file, just so we can load it back and compare.

Reply:
I'm definitely on board with that. I was thinking about trying to test for that, but wasn't sure if writing out the screenshot to a file was part of the example.

Reply:
It's a part of the example, but it doesn't mean that we should be testing that part. We can let PNG folks do their testing, and stdlib likewise with file output :)